spring/spring6/spring6-remote/src/test/java/org/infinispan/spring/remote/provider/SpringRemoteCacheTest.java

```diff
@@ -106,6 +108,44 @@ public void testValueLoaderWithLocking() throws Exception {
       assertEquals("thread1", valueObtainedByThread2);
    }
 
+   @Test(timeOut = 30_000)
+   public void testRetrieveMethods() throws Exception {
+      final SpringRemoteCacheManager springRemoteCacheManager = new SpringRemoteCacheManager(remoteCacheManager);
+      final SpringCache cache = springRemoteCacheManager.getCache(TEST_CACHE_NAME);
+
+      cache.put("test-one", "one");
+      String result = (String) ((Cache.ValueWrapper) cache.retrieve("test-one").get(1, TimeUnit.SECONDS)).get();
+      assertEquals("one", result);
+
+      result = cache.retrieve("test-two", () -> CompletableFuture.completedFuture("two")).get();
+      assertEquals("two", result);
+
+      CountDownLatch waitUntilThread1LocksValueGetter = new CountDownLatch(1);
```

Comment:
I guess a `CyclicBarrier` would fit better here instead. Otherwise, it is possible the countDown executes even before the second retrieve started.

Reply:
Ah, more tricky than that, I take. One of the suppliers should never execute.

Reply:
Yeah, I mentioned this to Katia earlier, but I would say it isn't a big deal as the other has the same problem and it should never be hit. (it would only be if the test failed flakily)

Reply:
The more I think about this, the way it is currently implemented is a possible flaky test failure.

Below there is nothing to guarantee the ordering of the lambdas. It is entirely possible the second runs before the first and sets the value to thread1. I don't recall why I thought this was okay before.

We do indeed need a CyclicBarrier to first verify the first lambda is inside before submitting the second retrieval operation.
```java
    CyclicBarrier barrier = new CyclicBarrier(2);

    Future<String> thread1 = fork(() -> cache.retrieve("test-3", () -> {
         barrier.arriveAndAwait(10, TimeUnit.SECONDS);
         barrier.arriveAndAwait(10, TimeUnit.SECONDS);
         return CompletableFuture.completedFuture("thread1");
      }).get());

      // Ensure first retrieval is in lambda before continuing
      barrier.arriveAndAwait(10, TimeUnit.SECONDS);

      Future<String> thread2 = fork(() -> cache.retrieve("test-3", () -> {
         return CompletableFuture.completedFuture("thread2");
      }).get());
      
      // Optionally add a short sleep here
      
      barrier.arriveAndAwait(10, TimeUnit.SECONDS);
```

Reply:
pushed the changes, in both methods